# Load package

In [1]:
import numpy as np
import matplotlib.pyplot as plt
%matplotlib notebook

# Data 

In [3]:
NPZ_outDir = '/home/group/Droplet/Results/NPZ/'
outDir = '/home/group/Droplet/Results/NPZ_Analysis/'


In [4]:
data = np.load(NPZ_outDir + 'CMS_20uLM_SB_data_updated_20230113.npz', allow_pickle=True)['df'][0]



In [5]:
print( data.keys() )
print( data['metadata'])

dict_keys(['metadata', 'uv', 'SAXS', 'MAXS'])
{'description': 'Au synthesis at CMS beamline 2022_3 cycle', 'temperature': '100 C', 'flow rate': '20 ul/min', 'date': '1013', 'chemical': ['NCit_16mM', 'Water with 0.01% Tween20', 'HCl(-) or NaCl(+) 10mM', 'HAuCl4_2mM'], 'volume_unit': 'ul'}


In [47]:
HA = data['SAXS']['data']['droplet_dict']['HA'] 
N = len(HA)
blue_droplets_index = np.arange(N)[HA==0 ]
Nb = len(blue_droplets_index)
print( 'There are totally %s droplets, including  %s blue dyes.'%(N, Nb))

There are totally 592 droplets, including  66 blue dyes.


In [58]:
iqb = data['SAXS']['data']['iqb'] 
droplet_without_saxs_data = []

for i, iqx in enumerate(iqb):
    if isinstance(iqx, float):
        droplet_without_saxs_data.append(i)
        
droplet_without_saxs_data

[77, 132, 140, 290, 299]

* NOTE:
** droplets are counts from zero
** there are five missing saxs data with index as  [77, 132, 140, 290, 299]

In [6]:
print(data['uv'].keys())

dict_keys(['description', 'raw_data', 'processed_old_data', 'processed_new_data', 'data'])


## SAXS

In [7]:
print( data['SAXS'].keys() )
print( data['SAXS']['data'].keys() )

dict_keys(['description', 'data'])
dict_keys(['droplet_filename_list', 'droplet_sid', 'droplet_dict', 'q', 'iq', 'qb', 'iqb'])


In [8]:
data['SAXS']['data']['iqb']

0      [0.39607187908508196, 0.3909230685951627, 0.40...
1      [0.353645984168431, 0.3228659743751829, 0.3040...
2      [0.36373422274855044, 0.36991118154505215, 0.3...
3      [2.856026650689201, 2.842350275946066, 2.83303...
4      [0.2367354198773948, 0.23445369123660384, 0.23...
                             ...                        
587    [1070.8666280110676, 1012.6724100748698, 965.7...
588    [916.998203531901, 861.2929829915364, 817.7403...
589    [1078.552891031901, 1017.1149881998698, 970.09...
590    [1011.9539978027344, 957.9382954915365, 915.75...
591    [1203.6061462402345, 1135.7981913248698, 1083....
Name: iqb, Length: 592, dtype: object

## UV

In [9]:
print( data['uv']['data'].keys()  )
print( data['uv']['data']['y']  )


dict_keys(['description', 'droplet_filename_list', 'droplet_dict', 'x_wav', 'y'])
0      [-0.012155581061388062, 0.006040977566928838, ...
1      [0.02065938387485666, -0.018680544828289358, 0...
2      [0.008809570115722249, -0.021829441293963834, ...
3      [0.1409325990029531, 0.1447077930553108, 0.127...
4      [0.015936111072974976, 0.01388884714877664, 0....
                             ...                        
587    [1.2980840982494266, 1.3086746904064541, 1.453...
588    [1.1331414704538811, 1.1987904991231022, 1.015...
589    [1.6939900147001328, 1.2436301563811725, 1.038...
590    [1.3994165903346076, 1.2379276455133175, 1.723...
591    [1.4317567444977843, 1.0582821445529673, 1.497...
Name: _y, Length: 592, dtype: object


# WAXS

In [10]:
print( data['MAXS']['data'].keys()  )
#print( data['MAXS']['data']['iq']  )

dict_keys(['droplet_filename_list', 'q', 'iq'])


In [11]:
#print( res['SAXS']['analysis']['216'] )

In [12]:
print( data['MAXS']['data']['q']  )

[0.34164825 0.34400141 0.34635457 ... 3.86432727 3.86668043 3.86903359]


In [13]:
print( len(data['MAXS']['data']['iq']  ))

592


# Analyssis results extracted from SAXS

In [19]:
res = np.load(NPZ_outDir +  'CMS_20uLM_SB_analysis_updated_20230123.npz', allow_pickle=True)['df'][0]


In [21]:
res.keys()

dict_keys(['metadata', 'SAXS'])

In [24]:
#res['SAXS']['analysis'].keys()
res['SAXS']['analysis']['0'].keys()

dict_keys(['q_iq', 'fit_res_init', 'fit_res_init_best', 'fit_res_init_best_paras'])

In [36]:
#data['SAXS']['data']

*  NOTE: radius_in_Angstron, sigma, maximun intensity, reduced error (mse)
 
 

In [59]:
res['SAXS']['analysis']['0']['fit_res_init_best_paras'][0]  #the first droplet

[185.0, 0.168, 0.05788357925219145, 0.07697589127667083]

In [60]:
res['SAXS']['analysis']['1']['fit_res_init_best_paras'][0]  #the first droplet

[482.0, 0.016, 0.029389944062721145, 0.11121178796330945]